clients/java/src/main/java/io/zeebe/client/impl/ZeebeClientCloudBuilderImpl.java

```diff
@@ -0,0 +1,168 @@
+  }
+
+  @Override
+  public ZeebeClientBuilder numJobWorkerExecutionThreads(final int numThreads) {
+    return innerBuilder.numJobWorkerExecutionThreads(numThreads);
+  }
+
+  @Override
+  public ZeebeClientBuilder defaultJobWorkerName(final String workerName) {
+    return innerBuilder.defaultJobWorkerName(workerName);
+  }
+
+  @Override
+  public ZeebeClientBuilder defaultJobTimeout(final Duration timeout) {
+    return innerBuilder.defaultJobTimeout(timeout);
+  }
+
+  @Override
+  public ZeebeClientBuilder defaultJobPollInterval(final Duration pollInterval) {
+    return innerBuilder.defaultJobPollInterval(pollInterval);
+  }
+
+  @Override
+  public ZeebeClientBuilder defaultMessageTimeToLive(final Duration timeToLive) {
+    return innerBuilder.defaultMessageTimeToLive(timeToLive);
+  }
+
+  @Override
+  public ZeebeClientBuilder defaultRequestTimeout(final Duration requestTimeout) {
+    return innerBuilder.defaultRequestTimeout(requestTimeout);
+  }
+
+  @Override
+  public ZeebeClientBuilder usePlaintext() {
+    return innerBuilder.usePlaintext();
+  }
+
+  @Override
+  public ZeebeClientBuilder caCertificatePath(final String certificatePath) {
+    return innerBuilder.caCertificatePath(certificatePath);
+  }
+
+  @Override
+  public ZeebeClientBuilder credentialsProvider(final CredentialsProvider credentialsProvider) {
+    return innerBuilder.credentialsProvider(credentialsProvider);
+  }
+
+  @Override
+  public ZeebeClientBuilder keepAlive(final Duration keepAlive) {
+    return innerBuilder.keepAlive(keepAlive);
+  }
+
+  @Override
+  public ZeebeClientBuilder withInterceptors(final ClientInterceptor... interceptor) {
+    return innerBuilder.withInterceptors(interceptor);
+  }
+
+  @Override
+  public ZeebeClientBuilder withJsonMapper(final JsonMapper jsonMapper) {
+    return innerBuilder.withJsonMapper(jsonMapper);
+  }
+
+  @Override
+  public ZeebeClient build() {
+    checkNotNull(clusterId, "clusterId can not be null");
+    checkNotNull(clientId, "clientId can not be null");
+    checkNotNull(clientSecret, "clientSecret can not be null");
+    final String cloudGatewayAddress = String.format("%s.%s:443", clusterId, BASE_ADDRESS);
+    innerBuilder.gatewayAddress(cloudGatewayAddress);
+    final OAuthCredentialsProviderBuilder builder = new OAuthCredentialsProviderBuilder();
+    final OAuthCredentialsProvider credentialsProvider =
+        builder
+            .audience(String.format("%s.%s", clusterId, BASE_ADDRESS))
+            .clientId(clientId)
+            .clientSecret(clientSecret)
+            .authorizationServerUrl(BASE_AUTH_URL)
+            .build();
+    innerBuilder.credentialsProvider(credentialsProvider);
+    return innerBuilder.build();
+  }
+
+  @Override
+  public String toString() {
+    final StringBuilder sb = new StringBuilder(innerBuilder.toString());
+    appendProperty(sb, "clusterId", clusterId);
+    appendProperty(sb, "clientId", clientId);
+    // TODO: Should we print clientSecret?
```

Comment:
No I think we should not print the secret :)

Reply:
Removed :)